api/samlacs.go

```diff
@@ -148,7 +148,22 @@ func (a *API) SAMLACS(w http.ResponseWriter, r *http.Request) error {
 		}
 	}
 
-	// TODO: fetch new metadata if possible when validUntil < time.Now() or cacheDuration
+	var samlMetadataModified bool
+	samlMetadataModified = false
+	if (!idpMetadata.ValidUntil.IsZero() && idpMetadata.ValidUntil.Before(time.Now())) || ssoProvider.SAMLProvider.UpdatedAt.Add(idpMetadata.CacheDuration).Before(time.Now()) {
+		if *ssoProvider.SAMLProvider.MetadataURL != "" {
+			rawMetadata, err := fetchSAMLMetadata(ctx, *ssoProvider.SAMLProvider.MetadataURL)
+			if err != nil {
+				// Fail silently but raise warning and continue with existing metadata
+				logentry := log.WithField("sso_provider_id", ssoProvider.ID.String())
+				logentry = logentry.WithField("saml_provider_metadata", ssoProvider.SAMLProvider.MetadataXML)
+				logentry.Warn("Metadata could not be retrieved. Continuing with existing metadata")
+			}
+			ssoProvider.SAMLProvider.MetadataXML = string(rawMetadata)
+			samlMetadataModified = true
```

Comment:
I believe this is a more correct version just in case.

```suggestion
			if err != nil {
				// Fail silently but raise warning and continue with existing metadata
				logentry := log.WithField("sso_provider_id", ssoProvider.ID.String())
				logentry = logentry.WithField("saml_provider_metadata", ssoProvider.SAMLProvider.MetadataXML)
				logentry.Warn("Metadata could not be retrieved. Continuing with existing metadata")
			} else {
				ssoProvider.SAMLProvider.MetadataXML = string(rawMetadata)
				samlMetadataModified = true
			}
```